app/assets/main/react_components/User/KanbanActionContainer.jsx

```diff
@@ -20,6 +20,40 @@ const KanbanActionContainer = ({ collapsed, setCollapsed, categories }) => {
   const setCategoryBadges = useCategoryBadgesUpdate();
   const setCategoryBadgesOnDrag = useCategoryBadgesUpdateOnDrag();
 
+  const handleExpanded = (item, value) => {
+    if (item.status === false) {
+      setColumns({
+        ...columns,
+        [1]: {
+          ...columns[1],
+          items: columns[1].items.map((expandable) => {
+            return expandable.id === item.id
+              ? {
+                  ...expandable,
+                  expanded: value,
+                }
+              : { ...expandable, expanded: false };
+          }),
+        },
+      });
+    } else {
+      setColumns({
+        ...columns,
+        [2]: {
+          ...columns[2],
+          items: columns[2].items.map((expandable) => {
+            return expandable.id === item.id
+              ? {
+                  ...expandable,
+                  expanded: value,
+                }
+              : { ...expandable, expanded: false };
+          }),
+        },
+      });
+    }
+  };
```

Comment:
One suggestion of refactoring would be to use a help function: 
  const handleExpanded = (item, value) => {
    if (item.status === false) {
      setColumns({
        ...columns,
        [1]: {
          ...columns[1],
          items: getExpandable(columns[1], item, value),
        },
      });
    } else {
      setColumns({
        ...columns,
        [2]: {
          ...columns[2],
          items: getExpandable(columns[2], item, value),
        },
      });
    }
  };

  const getExpandable = (column, item, value) => {
    const temp = column.items.map((expandable) => {
      return expandable.id === item.id
        ? {
            ...expandable,
            expanded: value,
          }
        : { ...expandable, expanded: false };
    });
    return temp;
  };

:D